### PR TITLE
WIP: [odo] Disconnected environments

### PR DESCRIPTION
**Pull request by ghost**
https://github.com/openshift/odo/issues/2189
https://github.com/openshift/odo/issues/2513
TODO: 
- java component on disconnected cluster
https://github.com/openshift/odo/issues/2514#issuecomment-575017489
- nodejs component on disconnected cluster
https://github.com/openshift/odo/issues/2478#issuecomment-574693844

### CODE REVIEW

File: cli_reference/openshift_developer_cli/odo-disconnected-env.adoc [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+* Certificate Authority (CA) certificate from the OpenShift cluster's host OS (`root` user)
+//link?
+// https://docs.openshift.com/container-platform/4.2/installing/install_config/installing-restricted-networks-preparations.html
+// TODO -> find docs how to setup a disconnected cluster
+// ^ yes they are there
+
+== Mirroring the `odo-init-image` into the cluster's local registry from the Red Hat registry
+//on the machine with cluster
+
+On the cluster's host OS:
```
Comment (kadel) [inline cli_reference/openshift_developer_cli/odo-disconnected-env.adoc]: This doesn't sound right. No one should ever do anything on cluster nodes! 
ping @amitkrout

File: cli_reference/openshift_developer_cli/odo-disconnected-env.adoc [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+----
+// host has to be conected to the internet
+// not an issue if "setting up the cluster for usage in restricted network" requires an internet connection as well
+. Mirror the `odo-init-image` image into the local mirror registry.
++
+----
+$ oc image mirror registry.access.redhat.com/openshiftdo/odo-init-image-rhel7:latest <mirror-registry>:<port>/openshiftdo/odo-init-image-rhel7:latest
+----
+
+// can you connect the (openshift) host machines to the internet? (not the cluster itself)
```
Comment (kadel) [inline cli_reference/openshift_developer_cli/odo-disconnected-env.adoc]: nope, that defeats the whole purpose of the disconnected cluster.

File: cli_reference/openshift_developer_cli/odo-disconnected-env.adoc [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
++
+----
+$ sudo cp ./disconnect-ca.crt /etc/pki/ca-trust/source/anchors/<mirror-registry>.crt
+----
++
+[NOTE]
+====
+CA certificate target paths may vary between operating systems.
+====
+// is the windows/macos commands the same?
```
Comment (kadel) [inline cli_reference/openshift_developer_cli/odo-disconnected-env.adoc]: they are not

File: _topic_map.yml [hunk 4]
```diff
@@ -1168,6 +1168,8 @@ Topics:
     File: creating-a-multicomponent-application-with-odo
   - Name: Creating an application with a database
     File: creating-an-application-with-a-database
+  - Name: Running odo in a disconnected environment
```
Comment (kadel) [inline _topic_map.yml]: The rest of the OCP docs call this "restricted networks" not "disconnected environment". We should use the same terminology

Comment (pmacko1) [reply _topic_map.yml]: yep, thanks!

File: cli_reference/openshift_developer_cli/odo-disconnected-env.adoc (hunk 2 above)
Comment (pmacko1) [reply cli_reference/openshift_developer_cli/odo-disconnected-env.adoc]: @amitkrout so we have to devise a workaround -> ssh from odo machine maybe?

File: cli_reference/openshift_developer_cli/odo-restricted-network.adoc [hunk 5]
```diff
@@ -0,0 +1,492 @@
+////
+Yana:
```
Comment (pmacko1) [inline cli_reference/openshift_developer_cli/odo-restricted-network.adoc]: @boczkowska

**Comment by amitkrout** [comment]
ping @boczkowska 
Let me know if you need any further clarification.

**Comment by girishramnani** [comment]
@boczkowska can you close this PR as we have a new PR for airgap https://github.com/openshift/openshift-docs/pull/20227

**Comment by amitkrout** [comment]
done via #20227
@boczkowska Can you please close this

**Comment by vikram-redhat** [comment]
Closing as per https://github.com/openshift/openshift-docs/pull/19128#issuecomment-641093046.

@Preeticp

**Comment by Preeticp** [comment]
Thank you @vikram-redhat 
Apologies, we did not get to it earlier.

**Pull request closed without merging**